src/mlpack/methods/ne/cne.hpp

```diff
@@ -0,0 +1,186 @@
+        genome.aLinkGenes[i].Weight(weight);
+      }
+    }
+  }
+
+  // Randomly select weights from one parent genome.
+  // NOTICE: child genomes need to set genome id based on its population's max id.
+  static void CrossoverWeights(Genome& momGenome, 
+                               Genome& dadGenome, 
+                               Genome& child1Genome, 
+                               Genome& child2Genome) {
+    child1Genome = momGenome;
+    child2Genome = dadGenome;
+    for (int i=0; i<momGenome.aLinkGenes.size(); ++i) { // assume genome are the same structure.
+      double t = mlpack::math::RandNormal();
+      if (t>0) {  // prob = 0.5
+        child1Genome.aLinkGenes[i].Weight(momGenome.aLinkGenes[i].Weight());
+        child2Genome.aLinkGenes[i].Weight(dadGenome.aLinkGenes[i].Weight());
+      } else {
+        child1Genome.aLinkGenes[i].Weight(dadGenome.aLinkGenes[i].Weight());
+        child2Genome.aLinkGenes[i].Weight(momGenome.aLinkGenes[i].Weight());
+      }
+    }
+  }
+
+  // Initializing the species of genomes.
+  // It can use species's parametric constructor.
+  // Besides, adapt its own style of initialization.
+  void InitSpecies() {
+    aSpecies = Species(aSeedGenome, aSpeciesSize);
+  }
+
+  // Reproduce the next species. Heart function for CNE !!!
+  // Select parents from G(i) based on their fitness.
+  // Apply search operators to parents and produce offspring which
+  // form G(i + 1).
+  void Reproduce() {
+    // Sort species by fitness
+    aSpecies.SortGenomes();
+
+    // Select parents from elite genomes and crossover.
+    int numElite = floor(aElitePercentage * aSpeciesSize);
+    int numDrop = floor((aSpeciesSize - numElite) / 2) * 2;  // Make sure even number.
+    numElite = aSpeciesSize - numDrop;
+    for (int i=numElite; i<aSpeciesSize-1; ++i) {
+      // Randomly select two parents from elite genomes.
+      int idx1 = RandInt(0, numElite);
+      int idx2 = RandInt(0, numElite);
+
+      // Crossover to get two children genomes.
+      CrossoverWeights(aSpecies.aGenomes[idx1], aSpecies.aGenomes[idx2],
+                               aSpecies.aGenomes[i], aSpecies.aGenomes[i+1]);
+    }
+
+    // Keep the best genome and mutate the rests.
+    for (int i=1; i<aSpeciesSize; ++i) {
+      MutateWeightsBiased(aSpecies.aGenomes[i], aMutateRate, aMutateSize);
+    }
+  }
+
+  // Evolution of species.
+  void Evolve() {
+    // Generate initial species at random.
+    int generation = 0;
+    InitSpecies();
+    
+    // Repeat
+    while (generation < aMaxGeneration) {
+    	// Evaluate all genomes in the species.
+      for (int i=0; i<aSpecies.SpeciesSize(); ++i) {
+        double fitness = aTask.EvalFitness(aSpecies.aGenomes[i]);
+        aSpecies.aGenomes[i].Fitness(fitness);
+      }
+      aSpecies.SetBestFitnessAndGenome();
+
+    	// Output some information.
+      printf("Generation: %zu\tBest fitness: %f\n", generation, aSpecies.BestFitness());
```

Comment:
I think it would be better to use Log::Info here, in that case a user can disable any output e.g:

```
Log::Info << "Generation: " << generation << " best fitness: " <<  species.BestFitness() << std::endl
```

or something like that, the important thing is a user can disable the output.